src/natcap/invest/datastack.py

```diff
@@ -311,77 +192,217 @@ def build_datastack_archive(args, model_name, datastack_path):
-
-                args_key_filter = _ArgsKeyFilter(args_key_label)
-                handler.addFilter(args_key_filter)
-                if args_key not in ('workspace_dir',):
-                    new_dict[args_key] = _recurse(args_value, handler,
-                                                  nested_key=args_key_label)
-                handler.removeFilter(args_key_filter)
-            return new_dict
-        elif isinstance(args_param, str):
-            # If the parameter string is blank, return an empty string.
-            if args_param.strip() == '':
-                return ''
-
-            # It's a string and exists on disk, it's a file!
-            possible_path = os.path.normpath(args_param.replace('\\', os.sep))
-            if os.path.exists(possible_path):
-                try:
-                    filepath = files_found[possible_path]
-                    LOGGER.debug(('Parameter known from a previous '
-                                  'entry: %s, using %s'),
-                                 possible_path, filepath)
-                    return filepath
-                except KeyError:
-                    # turn the nested key into a nice name for a folder
-                    # e.g. args['category']['data_path'] --> category_data_path
-                    folder_prefix = nested_key[6 : -2].replace('\'][\'', '_')
-                    found_filepath = _collect_filepath(possible_path,
-                                                       data_dir,
-                                                       folder_prefix)
-
-                    # Store only linux-style filepaths.
-                    relative_filepath = os.path.relpath(
-                        found_filepath, temp_workspace).replace('\\', '/')
-                    files_found[possible_path] = relative_filepath
-                    LOGGER.debug('Processed path %s to %s',
-                                 args_param, relative_filepath)
-                    return relative_filepath
-        # It's not a file or a structure to recurse through, so
-        # just return the item verbatim.
-        LOGGER.info('Using verbatim value: %s', args_param)
-        return args_param
-
-    log_format = "%(args_key)-25s %(name)-25s %(levelname)-8s %(message)s"
-    with utils.log_to_file(logfile, log_fmt=log_format) as handler:
-        new_args = {
-            'args': _recurse(args, handler),
-            'model_name': model_name,
-            'invest_version': __version__
-        }
+                    shutil.copyfile(src_path, dest_path)
+
+            LOGGER.debug(
+                f'Directory copied from {source_path} --> {target_directory}')
+            target_arg_value = _relpath(target_directory)
+            files_found[source_path] = target_arg_value
+
+        elif input_type in spatial_types:
+            # Create a directory with a readable name, something like
+            # "aoi_path_vector" or "lulc_cur_path_raster".
```

Comment:
I'm curious if there was a functional reason for making a directory for each spatial file? Or is that just the way you prefer to keep things organized? I agree it's nice for organization, especially with shapefiles or ArcGrids floating around. I'm not suggesting a change.

Reply:
Yeah, it comes down to those multipart files.  It's not a big deal with ESRI Shapefiles because all of the component files have the vector's basename (`watersheds.shp`, `watersheds.dbf`, etc.).  Arc Binary Grids on the other hand are a known problem because the raster's name is the parent directory but the component files are all pretty much the same as another Arc Binary Grid.  So if a user happened to provide two Arc Binary Grid rasters and we copied them into the data directory, we'd have a conflict, not to mention that we can't just rename `hdr.adf` to `lulc_cur_path` and expect the raster to work!

So given how many file formats GDAL supports, putting all the component files into a directory just seemed the most flexible solution than trying to handle a special case for those raster formats that are a little stranger.